The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8484603926/job/23247851374) [(plain)](https://github.com/rust-lang/rust/commit/c6ac3b02dbd74a4116f7198c442f419fee3cacc5/checks/23247851374/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:b4ffde65f46336ab88eb53be808477a3936bae11)
Download action repository 'actions/upload-artifact@v4' (SHA:5d5d22a31266ced268874388b861e4b58bb5c2f3)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.518   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.537 Collecting boolean-py==4.0
#13 0.547   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.568 Collecting chardet==5.1.0
---
#13 3.739 Building wheels for collected packages: reuse
#13 3.740   Building wheel for reuse (pyproject.toml): started
#13 4.061   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 4.062   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 4.062   Stored in directory: /tmp/pip-ephem-wheel-cache-icccie1n/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 4.065 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 4.087   Attempting uninstall: setuptools
#13 4.088     Found existing installation: setuptools 59.6.0
#13 4.089     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 4.089     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 4.089     Can't uninstall 'setuptools'. No files were found to uninstall.
#13 4.755 Successfully installed binaryornot-0.4.4 boolean-py-4.0 chardet-5.1.0 jinja2-3.1.2 license-expression-30.0.0 markupsafe-2.1.1 python-debian-0.1.49 reuse-1.1.0 setuptools-66.0.0
#13 4.755 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 5.276 Collecting virtualenv
#13 5.340   Downloading virtualenv-20.25.1-py3-none-any.whl (3.8 MB)
#13 5.561      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.8/3.8 MB 17.2 MB/s eta 0:00:00
#13 5.615 Collecting platformdirs<5,>=3.9.1
#13 5.625   Downloading platformdirs-4.2.0-py3-none-any.whl (17 kB)
#13 5.660 Collecting filelock<4,>=3.12.2
#13 5.670   Downloading filelock-3.13.3-py3-none-any.whl (11 kB)
#13 5.692 Collecting distlib<1,>=0.3.7
#13 5.703   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.719      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 32.2 MB/s eta 0:00:00
#13 5.804 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.964 Successfully installed distlib-0.3.8 filelock-3.13.3 platformdirs-4.2.0 virtualenv-20.25.1
#13 DONE 6.0s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      182208 kB
DirectMap2M:     7157760 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished `dev` profile [unoptimized] target(s) in 0.03s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/399fa2f6e419fd2c70942c191c2e55814af8d167/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-399fa2f6e419fd2c70942c191c2e55814af8d167-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 25.79s
##[endgroup]
fmt check
##[error]Diff in /checkout/library/core/src/task/wake.rs at line 339:
     pub const fn from_waker(waker: &'a Waker) -> Self {
         // SAFETY: LocalWaker is just Waker without thread safety
         let local_waker = unsafe { transmute(waker) };
-        Self { waker: waker, local_waker, ext: ExtData::None(()), _marker: PhantomData, _marker2: PhantomData }
+            waker: waker,
+            local_waker,
+            ext: ExtData::None(()),
+            _marker: PhantomData,
+            _marker: PhantomData,
+            _marker2: PhantomData,
+        }
     }
 
     /// Create a ContextBuilder from an existing Context.
##[error]Diff in /checkout/library/core/src/task/wake.rs at line 351:
             ExtData::Some(ext) => ExtData::Some(*ext),
             ExtData::None(()) => ExtData::None(()),
         };
-        Self { waker: cx.waker, local_waker: cx.local_waker, ext, _marker: PhantomData, _marker2: PhantomData }
+        Self {
+            waker: cx.waker,
+            local_waker: cx.local_waker,
+            _marker: PhantomData,
+            _marker2: PhantomData,
+        }
     }
     }
 
     /// This method is used to set the value for the waker on `Context`.
Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/core/src/convert/mod.rs" "/checkout/library/core/src/convert/num.rs" "/checkout/library/core/src/lib.rs" "/checkout/library/core/src/array/equality.rs" "/checkout/library/core/src/array/ascii.rs" "/checkout/library/core/src/array/mod.rs" "/checkout/library/core/src/array/drain.rs" "/checkout/library/core/src/array/iter.rs" "/checkout/library/core/src/iter/sources/from_fn.rs" "/checkout/library/core/src/iter/sources/repeat_n.rs" "/checkout/library/core/src/iter/sources/successors.rs" "/checkout/library/core/src/iter/sources/repeat.rs" "/checkout/library/core/src/iter/sources/once_with.rs" "/checkout/library/core/src/iter/sources/empty.rs" "/checkout/library/core/src/iter/sources/from_coroutine.rs" "/checkout/library/core/src/iter/sources/once.rs" "/checkout/library/core/src/iter/sources/repeat_with.rs" "/checkout/library/core/src/alloc/global.rs" "/checkout/library/core/src/alloc/mod.rs" "/checkout/library/core/src/alloc/layout.rs" "/checkout/library/core/src/str/lossy.rs" "/checkout/library/core/src/str/error.rs" "/checkout/library/core/src/str/mod.rs" "/checkout/library/core/src/str/converts.rs" "/checkout/library/core/src/str/count.rs" "/checkout/library/core/src/str/traits.rs" "/checkout/library/core/src/str/iter.rs" "/checkout/library/core/src/str/validations.rs" "/checkout/library/core/src/str/pattern.rs" "/checkout/library/core/src/ascii.rs" "/checkout/library/core/src/result.rs" "/checkout/library/core/src/error.rs" "/checkout/library/core/src/sync/exclusive.rs" "/checkout/library/core/src/sync/atomic.rs" "/checkout/library/core/src/sync/mod.rs" "/checkout/library/core/src/task/ready.rs" "/checkout/library/core/src/task/mod.rs" "/checkout/library/core/src/task/wake.rs" "/checkout/library/core/src/task/poll.rs" "/checkout/library/core/src/clone.rs" "/checkout/library/core/src/pin.rs" "/checkout/library/core/src/primitive_docs.rs" "/checkout/library/core/src/tuple.rs" "/checkout/library/core/src/slice/rotate.rs" "/checkout/library/core/src/slice/select.rs" "/checkout/library/core/src/slice/raw.rs" "/checkout/library/core/src/slice/ascii.rs" "/checkout/library/core/src/slice/mod.rs" "/checkout/library/core/src/slice/memchr.rs" "/checkout/library/core/src/slice/sort.rs" "/checkout/library/core/src/async_iter/mod.rs" "/checkout/library/core/src/async_iter/async_iter.rs" "/checkout/library/core/src/async_iter/from_iter.rs" "/checkout/library/core/src/intrinsics.rs" "/checkout/library/core/src/slice/iter/macros.rs" "/checkout/library/core/src/slice/index.rs" "/checkout/library/core/src/slice/iter.rs" "/checkout/library/core/src/slice/specialize.rs" "/checkout/library/core/src/slice/cmp.rs" "/checkout/library/core/src/cell.rs" "/checkout/library/core/src/cmp/bytewise.rs" "/checkout/library/core/src/unicode/mod.rs" "/checkout/library/core/src/unicode/printable.rs" "/checkout/library/core/src/iter/sources.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Fri Mar 29 19:12:09 UTC 2024
  network time: Fri, 29 Mar 2024 19:12:09 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/task/wake.rs
+++ b/library/core/src/task/wake.rs
@@ -276,7 +276,7 @@ impl<'a> Context<'a> {
 
     /// Returns a reference to the extension data for the current task.
     #[inline]
-    #[unstable(feature = "context_ext", issue = "none")]
+    #[unstable(feature = "context_ext", issue = "123392")]
     #[rustc_const_unstable(feature = "const_waker", issue = "102012")]
     pub const fn ext(&mut self) -> &mut dyn Any {
         match &mut self.ext {
@@ -351,7 +351,7 @@ impl<'a> ContextBuilder<'a> {
     /// Create a ContextBuilder from an existing Context.
     #[inline]
     #[rustc_const_unstable(feature = "const_waker", issue = "102012")]
-    #[unstable(feature = "context_ext", issue = "none")]
+    #[unstable(feature = "context_ext", issue = "123392")]
     pub const fn from(cx: &'a mut Context<'_>) -> Self {
         let ext = match &mut cx.ext {
             ExtData::Some(ext) => ExtData::Some(*ext),
@@ -368,7 +368,7 @@ impl<'a> ContextBuilder<'a> {
 
     /// This method is used to set the value for the waker on `Context`.
     #[inline]
-    #[unstable(feature = "context_ext", issue = "none")]
+    #[unstable(feature = "context_ext", issue = "123392")]
     #[rustc_const_unstable(feature = "const_waker", issue = "102012")]
     pub const fn waker(self, waker: &'a Waker) -> Self {
         Self { waker, ..self }
@@ -384,7 +384,7 @@ impl<'a> ContextBuilder<'a> {
 
     /// This method is used to set the value for the extension data on `Context`.
     #[inline]
-    #[unstable(feature = "context_ext", issue = "none")]
+    #[unstable(feature = "context_ext", issue = "123392")]
     #[rustc_const_unstable(feature = "const_waker", issue = "102012")]
     pub const fn ext(self, data: &'a mut dyn Any) -> Self {
         Self { ext: ExtData::Some(data), ..self }

--- a/library/core/src/task/wake.rs
+++ b/library/core/src/task/wake.rs
@@ -339,7 +339,13 @@ impl<'a> ContextBuilder<'a> {
     pub const fn from_waker(waker: &'a Waker) -> Self {
         // SAFETY: LocalWaker is just Waker without thread safety
         let local_waker = unsafe { transmute(waker) };
-        Self { waker: waker, local_waker, ext: ExtData::None(()), _marker: PhantomData, _marker2: PhantomData }
+        Self {
+            waker: waker,
+            local_waker,
+            ext: ExtData::None(()),
+            _marker: PhantomData,
+            _marker2: PhantomData,
+        }
     }
 
     /// Create a ContextBuilder from an existing Context.
@@ -351,7 +357,13 @@ impl<'a> ContextBuilder<'a> {
             ExtData::Some(ext) => ExtData::Some(*ext),
             ExtData::None(()) => ExtData::None(()),
         };
-        Self { waker: cx.waker, local_waker: cx.local_waker, ext, _marker: PhantomData, _marker2: PhantomData }
+        Self {
+            waker: cx.waker,
+            local_waker: cx.local_waker,
+            ext,
+            _marker: PhantomData,
+            _marker2: PhantomData,
+        }
     }
 
     /// This method is used to set the value for the waker on `Context`.

--- a/tests/ui/async-await/async-is-unwindsafe.rs
+++ b/tests/ui/async-await/async-is-unwindsafe.rs
@@ -11,6 +11,7 @@ fn main() {
 
     is_unwindsafe(async {
         //~^ ERROR the type `&mut Context<'_>` may not be safely transferred across an unwind boundary
+        //~| ERROR the type `&mut (dyn Any + 'static)` may not be safely transferred across an unwind boundary
         use std::ptr::null;
         use std::task::{Context, RawWaker, RawWakerVTable, Waker};
         let waker = unsafe {

--- a/tests/ui/async-await/async-is-unwindsafe.stderr
+++ b/tests/ui/async-await/async-is-unwindsafe.stderr
@@ -6,19 +6,18 @@ LL |        is_unwindsafe(async {
    |  |_____|
    | ||
 LL | ||
+LL | ||
 LL | ||         use std::ptr::null;
-LL | ||         use std::task::{Context, RawWaker, RawWakerVTable, Waker};
 ...  ||
 LL | ||         drop(cx_ref);
 LL | ||     });
    | ||_____-^ `&mut Context<'_>` may not be safely transferred across an unwind boundary
    |  |_____|
-   |        within this `{async block@$DIR/async-is-unwindsafe.rs:12:19: 29:6}`
+   |        within this `{async block@$DIR/async-is-unwindsafe.rs:12:19: 30:6}`
    |
-   = help: within `{async block@$DIR/async-is-unwindsafe.rs:12:19: 29:6}`, the trait `UnwindSafe` is not implemented for `&mut Context<'_>`, which is required by `{async block@$DIR/async-is-unwindsafe.rs:12:19: 29:6}: UnwindSafe`
-   = note: `UnwindSafe` is implemented for `&Context<'_>`, but not for `&mut Context<'_>`
+   = help: within `{async block@$DIR/async-is-unwindsafe.rs:12:19: 30:6}`, the trait `UnwindSafe` is not implemented for `&mut Context<'_>`, which is required by `{async block@$DIR/async-is-unwindsafe.rs:12:19: 30:6}: UnwindSafe`
 note: future does not implement `UnwindSafe` as this value is used across an await
-  --> $DIR/async-is-unwindsafe.rs:25:18
+  --> $DIR/async-is-unwindsafe.rs:26:18
    |
 LL |         let cx_ref = &mut cx;
    |             ------ has type `&mut Context<'_>` which does not implement `UnwindSafe`
@@ -31,6 +30,38 @@ note: required by a bound in `is_unwindsafe`
 LL | fn is_unwindsafe(_: impl std::panic::UnwindSafe) {}
    |                          ^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `is_unwindsafe`
 
-error: aborting due to 1 previous error
+error[E0277]: the type `&mut (dyn Any + 'static)` may not be safely transferred across an unwind boundary
+  --> $DIR/async-is-unwindsafe.rs:12:5
+   |
+LL |        is_unwindsafe(async {
+   |   _____^_____________-
+   |  |_____|
+   | ||
+LL | ||
+LL | ||
+LL | ||         use std::ptr::null;
+...  ||
+LL | ||         drop(cx_ref);
+LL | ||     });
+   | ||_____-^ `&mut (dyn Any + 'static)` may not be safely transferred across an unwind boundary
+   |  |_____|
+   |        within this `{async block@$DIR/async-is-unwindsafe.rs:12:19: 30:6}`
+   |
+   = help: within `{async block@$DIR/async-is-unwindsafe.rs:12:19: 30:6}`, the trait `UnwindSafe` is not implemented for `&mut (dyn Any + 'static)`, which is required by `{async block@$DIR/async-is-unwindsafe.rs:12:19: 30:6}: UnwindSafe`
+note: future does not implement `UnwindSafe` as this value is used across an await
+  --> $DIR/async-is-unwindsafe.rs:26:18
+   |
+LL |         let mut cx = Context::from_waker(&waker);
+   |             ------ has type `Context<'_>` which does not implement `UnwindSafe`
+...
+LL |         async {}.await; // this needs an inner await point
+   |                  ^^^^^ await occurs here, with `mut cx` maybe used later
+note: required by a bound in `is_unwindsafe`
+  --> $DIR/async-is-unwindsafe.rs:3:26
+   |
+LL | fn is_unwindsafe(_: impl std::panic::UnwindSafe) {}
+   |                          ^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `is_unwindsafe`
+
+error: aborting due to 2 previous errors
 
 For more information about this error, try `rustc --explain E0277`.

